tests/integration/tests/test_version_upgrades.py

```diff
@@ -175,3 +176,24 @@ def test_version_downgrades_with_rollback(instances: List[harness.Instance], tmp
         LOG.info("Rollback segment complete. Proceeding to next downgrade segment.")
 
     LOG.info("Rollback test complete. All downgrade segments verified.")
+
+
+@pytest.mark.tags(tags.NIGHTLY)
+def test_feature_upgrades(instances: List[harness.Instance]):
+    # Test the feature upgrades work
+    # Note(ben): This test is a work in progress and will
+    # be expanded with new tests as the feature upgrades work takes shape.
+    # Once this work is complete, this test will likely be merged with the
+    # test_version_upgrades test above and create a single test for all upgrades.
```

Comment:
super nit: I think this is a function doc:
```suggestion
    """Test the feature upgrades work
    Note(ben): This test is a work in progress and will
    be expanded with new tests as the feature upgrades work takes shape.
    Once this work is complete, this test will likely be merged with the
    test_version_upgrades test above and create a single test for all upgrades."""
```